src/Symfony/Component/HttpKernel/Tests/KernelTest.php

```diff
@@ -761,6 +761,21 @@ public function testSymfonyEnvironmentVariables()
         unset($_SERVER['SYMFONY__FOO__BAR']);
     }
 
+    public function testProjectDirExtension()
+    {
+        $kernel = $this->getKernel(array('initializeBundles', 'initializeContainer', 'getProjectDir'));
+        $kernel
+            ->expects($this->atLeastOnce())
+            ->method('getProjectDir')
+            ->willReturn('foo');
+        $kernel->boot();
+
+        $r = new \ReflectionMethod($kernel, 'getKernelParameters');
+        $r->setAccessible(true);
```

Comment:
Besides due the constructor initialization this test also passes without your change :) so we actually need to test some weird edge case here.